actors/builtin/miner/miner_actor.go

```diff
@@ -102,7 +102,25 @@ func (a Actor) Constructor(rt Runtime, params *ConstructorParams) *adt.EmptyValu
 	periodStart := nextProvingPeriodStart(currEpoch, offset)
 	Assert(periodStart > currEpoch)
 
-	state, err := ConstructState(emptyArray, emptyMap, emptyDeadlinesCid, owner, worker, params.PeerId, params.Multiaddrs, params.SealProofType, periodStart)
+	sectorSize, err := params.SealProofType.SectorSize()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "no sector size for seal proof type %d: %w", params.SealProofType)
+
+	partitionSectors, err := params.SealProofType.WindowPoStPartitionSectors()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "no partition size for seal proof type %d", params.SealProofType)
+
+	info := MinerInfo{
+		Owner:                      owner,
+		Worker:                     worker,
+		PendingWorkerKey:           nil,
+		PeerId:                     params.PeerId,
+		Multiaddrs:                 params.Multiaddrs,
+		SealProofType:              params.SealProofType,
+		SectorSize:                 sectorSize,
+		WindowPoStPartitionSectors: partitionSectors,
+	}
```

Comment:
Please make a `ConstructMinerInfo` function in miner_state.go to push a little of the noise into there.